cybersecurity/port-scanner/devashmit-day17/port_scanner.py

```
@@ -0,0 +1,39 @@
+ETHICS: Only scan systems you own or have permission to scan."""
+import socket, threading
+from datetime import datetime
+
+SERVICES={21:"FTP",22:"SSH",23:"Telnet",25:"SMTP",53:"DNS",80:"HTTP",110:"POP3",
+          143:"IMAP",443:"HTTPS",445:"SMB",3306:"MySQL",3389:"RDP",5432:"PostgreSQL",
+          6379:"Redis",8080:"HTTP-Alt",8443:"HTTPS-Alt",27017:"MongoDB"}
+open_ports=[]; lock=threading.Lock()
+
+def scan(target, port):
+    try:
+        s=socket.socket(socket.AF_INET,socket.SOCK_STREAM); s.settimeout(0.5)
```

Ruff (0.15.15)

[error] Multiple statements on one line (semicolon)

(E702)